src/util/http/HttpClient.cpp

```diff
@@ -120,3 +121,20 @@ std::istringstream HttpClientImpl<StreamType>::sendRequest(
 // Explicit instantiations for HTTP and HTTPS, see the bottom of `HttpClient.h`.
 template class HttpClientImpl<beast::tcp_stream>;
 template class HttpClientImpl<ssl::stream<tcp::socket>>;
+
+// ____________________________________________________________________________
+std::istringstream sendHttpOrHttpsRequest(
+    ad_utility::httpUtils::Url url, const boost::beast::http::verb& method,
+    std::string_view requestData, std::string_view contentTypeHeader,
+    std::string_view acceptHeader) {
+  auto sendRequest = [&]<typename Client>() {
+    Client client{url.host(), url.port()};
+    return client.sendRequest(method, url.host(), url.target(), requestData,
+                              contentTypeHeader, acceptHeader);
+  };
+  if (url.protocol() == ad_utility::httpUtils::Url::Protocol::HTTP) {
+    return sendRequest.operator()<HttpClient>();
+  } else {
+    return sendRequest.operator()<HttpsClient>();
+  }
+}
```

Comment:
I think using your HTTP test  infrastructure, you could test at least the HTTP case, this would make the coverage happy.